src/org/openj9/envInfo/MachineInfo.java

```diff
@@ -38,6 +38,7 @@ public class MachineInfo {
 	public static final String[] CPU_CORES_CMD_SOLARIS = new String[] {"bash", "-c", "psrinfo | wc -l"};
 	public static final String[] NUMA_CMD = new String[] {"bash", "-c", "numactl --show | grep 'No NUMA support available on this system"};
 	public static final String[] SYS_VIRT_CMD = new String[] {""};
+	public static final String[] CHECK_DOCKER_CMD = new String[] {"bash", "-c", "if grep -sq 'docker\\|lxc' /proc/1/cgroup; then echo 'running on docker'; else echo 'not on docker'; fi"};
```

Comment:
I'm not sure what this check is supposed to do, but If I'm reading this right it looks for `docker` or `lxc` in file `/proc/1/cgroup`? Is this supposed to be an or match? If so, isn't `-E` missing on the grep? Either way, this seems a strange thing to do. Why not use `.dockerenv`? See: https://github.com/moby/libnetwork/pull/815/commits/66ab744d19e534ed723fdb7b8df11a5e95f05630

While this would still not work if the container engine is `podman`, it would be a more "standard" approach. @smlambert thoughts?

Reply:
yep, the purpose of this check is to tell us where we are running... bare metal or in a container, happy to update it to a more standard approach

Reply:
The thing is that it's not easily detectable for generic containers. If you care about `docker` only, using `.dockerenv` is the better thing to do. It makes it clear that this won't work generically, yet will return the right thing if run on `docker`. It won't work on podman or criu for example, though (but I don't think the current impl does either).